### PR TITLE
Toolmetrics

### PR DESCRIPTION
**Pull request by knorris**

### CODE REVIEW

File: src/js/components/Analytics.jsx [hunk 1]
```diff
@@ -43,9 +45,17 @@ class Analytics extends Component {
...
+    const event = {
+      group: 'pageView',
+      attributes: {
+        page: 'Analytics',
+        user: user.data.username
+      }
+    }
 
     dispatch(fetchSources())
+    dispatch(sendMetrics([event]))
```
Comment (pml984) [inline src/js/components/Analytics.jsx]: We may be able to move this call and the others that are just tracking page views into the onEnter hook of the Router in index.jsx.

https://github.com/reactjs/react-router/blob/master/examples/auth-flow/app.js#L133

**Comment by jeffshaver** [comment]
please rebase with dev

File: src/js/components/Analytics.jsx (hunk 1 above)
Comment (jeffshaver) [inline src/js/components/Analytics.jsx]: I am going to agree with @pml984 here. I don't want to have to modify each page to add the metrics. It should just happen when the routes change.

**Comment by jeffshaver** [comment]
@pml984 re-review this. ignore the travis-ci push failure. idk why its choking on that one, but its fine.

**Comment by jeffshaver** [comment]
@knorris rebase with dev and squash the commits and travis should pass your build.

**Comment by jeffshaver** [comment]
Overall I think this is much cleaner than the original approach. The one question I have is why every component knows about `user` now? Since you are running `@connect` on the LogMetrics component, you could just do something like this probably:

``` javascript
@connect(({user}) => ({user}))
export const LogMetrics = (groupName, displayName) => {
  // ...code
}
```

**Comment by knorris** [comment]
OK. Did not know that.

**Pull request merged**